examples/vault/vault.yml

```diff
@@ -0,0 +1,9 @@
+quarkus:
+  file:
+    vault:
+      provider:
+        pim:
```

Comment:
same as basic-auht-yaml file, shouldn't we need to add another entry matching with the below `credentials-provider`? such as `quarkus.file.vault.provider.pim.dbpwd`?